api/controllers/v0.3/ILSHelper.js

```diff
@@ -0,0 +1,390 @@
+/* eslint-disable */
+
+/**
+ * Client that connects to the ILS.
+ */
+class ILSAPIClient {}
+
+/**
+ * Helper class to setup API calls to the ILS.
+ */
+class IlsHelper {
+  // TODO: Implement the error classes and codes
+  // const StandardError = (err = "Standard error") => { throw new Error(err) };
+  // const IlsError = () => StandardError("ILS Error");
+  // const NotFoundError = () => IlsError();
+  // const MultipleMatchesError = () => IlsError();
+  // const HttpError = () => IlsError();
+  // const ConnectionTimeoutError = () => HttpError();
+
+  constructor(ilsClient = undefined) {
+    this.client =
+      ilsClient ||
+      new ILSAPIClient({
+        wsdl: this.ilsUrl,
+        open_timeout: 15,
+        read_timeout: 15,
+        log: true,
+        filters: ["username", "password"],
+        log_level: "debug",
+        pretty_print_xml: true,
+      });
+  }
+
+  ilsUrl() {
+    return `https://${process.env["ILS_SERVER"]}/iii/patronio/services/PatronIO?wsdl`;
+  }
+
+  callAndFailGracefully(params) {
+    try {
+      this.client.call(params);
+    } catch (e) {
+      let error;
+      switch (
+        e["fault"]["detail"]["com_iii_webapps_ws_patronio_patron_io_fault"][
+          "code"
+        ]
+      ) {
+        case IlsHelper.NOT_FOUND:
+          error = "NotFoundError()";
+          break;
+        case IlsHelper.MULTIPLE_MATCHES:
+          error = "MultipleMatchesError()";
+          break;
+        default:
+          error = "IlsError()";
+      }
+      throw new Error(error);
+      // TODO: catch HTTP timout errors
+      // These errors are returned when the SOAP client times out
+      // when connecting and when requesting (respectively).
+    }
+  }
+
+  search(query) {
+    return this.callAndFailGracefully({
+      type: "search_patrons",
+      message: {
+        search_string: query,
+        username: process.env["ILS_USERNAME"],
+        password: process.env["ILS_PASSWORD"],
+      },
+    });
+  }
+
+  create_patron(patron) {
+    if (!patron.validForIls) {
```

Comment:
What does it mean for a patron to be valid for ILS--can we get a comment?

Reply:
That can be found in the Card model code (Card and Patron are interchangeable in the ruby code and makes this confusing). But it's valid for the ILS if the patron data is valid and there's a ptype. I'll write more comments.